sockeye/train.py

```diff
@@ -399,6 +400,16 @@ def main():
         else:
             # Making MXNet module API's default scaling factor explicit
             optimizer_params["rescale_grad"] = 1.0 / args.batch_size
+        if optimizer == C.OPTIMIZER_EVE:
+            check_condition(args.normalize_loss, "Eve optimizer requires loss normalization (use --normalize-loss)")
+            check_condition(not clip_gradient,
+                            "Eve optimizer does not support gradient clipping (use --clip-gradient -1)")
+            optimizer_params["use_batch_objective"] = args.eve_loss in(C.EVE_LOSS_BATCH, C.EVE_LOSS_BOTH)
+            optimizer_params["use_checkpoint_objective"] = args.eve_loss in(C.EVE_LOSS_CHECKPOINT, C.EVE_LOSS_BOTH)
+        # Manually specified params
+        if args.optimizer_params:
```

Comment:
args.optimizer_params is just a dictionary, right?
You can to optimizer_params.update(args.optimizer_params) then